src/dotnet-interactive-vscode/syntaxes/source.dotnet-interactive.magic-commands.tmGrammar.json

```diff
@@ -1,9 +1,63 @@
 {
+    "name": ".NET Interactive Magic Commands",
     "scopeName": "source.dotnet-interactive.magic-commands",
     "patterns": [
         {
-            "name": "comment.magic-command",
-            "match": "^#!(?!([cf](#|s(harp)?)|powershell|pwsh|html|javascript|js|markdown|md)).*$"
+            "name": "comment.line.magic-commands",
+            "match": "^#!(?!([cf](#|s(harp)?)|powershell|pwsh|html|javascript|js|markdown|md).*$)"
+        },
+        {
+            "include": "#magic-command-name"
+        },
+        {
+            "include": "#flag-argument"
+        },
+        {
+            "include": "#parameter"
+        },
+        {
+            "include": "#strings"
+        }
+    ],
+    "repository": {
+        "magic-command-name": {
+            "patterns": [
+                {
+                    "name": "keyword.control.magic-commands",
+                    "match": "(?<=^#!)[a-zA-Z0-9_-]+"
+                }
+            ]
+        },
+        "flag-argument": {
```

Comment:
In GNU/POSIX command line terminology and thus in System.CommandLine this is an "option". The Windows equivalent includes e.g. `/p` so you should add `/` as a prefix to this regex.